lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -709,6 +758,12 @@ impl Instance {
             module,
             ctx: Context::new(),
             state: State::Ready,
+            kill_state: Arc::new(KillState {
+                terminable: AtomicBool::new(false),
+                tid_change_notifier: Condvar::new(),
+                execution_domain: Mutex::new(Domain::Guest),
+                thread_id: Mutex::new(None),
+            }),
```

Comment:
Since there are no free variables here, I'd factor this out into something like `KillState::new_arc()`